Documentation/ceph-bucket-notification-crd.md

```diff
@@ -0,0 +1,56 @@
+# Ceph bucket notifications CRD
+
+## Overview
+
+In its new release, Ceph introduced the bucket notifications feature. It allows to send messages to various endpoints when a new event occurs on a bucket (ref: <https://docs.ceph.com/docs/master/radosgw/notifications/)>
+
+Setup of those notifications are normally done by sending HTTP requests to the RGW, either to create/delete topics pointing to specific endpoints, or create/delete bucket notifications based on those topics.
+
+This functionality eases this process by avoiding to use external tools or scripts. It is replaced by creation of CR definitions that  contain all the information necessary to create topics and/or notifications, which the rook operator processes.
+
+## Goals
+
+Creates a CRD for topics and a CRD for notifications, defining all the necessary and optional information for the various endpoints.
+
+Extends the rook operator to handle the CRs that would be submitted by users.
```

Comment:
For OBC, we can utilize  additionalConfig
```yaml
spec:
   AdditionalConfig:
              bucketnotification: notification1, notification2 
```

Reply:
Now that's interesting! I'll make a proposal around this.
But also we cannot be sure that people will always use OBCs to create buckets for various reasons (automated s3cmd scripts or alike,...). So in any case I guess a separate CRD would still be useful. I guess both approach can co-exist.